### ISSUE TITLE
Create BEM Style Guide

### ISSUE THREAD

**Issue by sfrisk** [jquery/css-chassis#77]
As we've decided to go with a BEM naming convention, to help people easily contribute to the Chassis project, we need a BEM Style Guide to help understanding the naming convention as we're using it.

**Comment by sfrisk** [comment]
Going along with the BEM Style Guide, here are some examples of the "flavor" of BEM we decided on:

`.ui_base__some-element--modifier--another-modifier`
`.ui-button__icon--rainbow-mode`

**Comment by kristyjy** [comment]
Since we aren't using BEM anymore should we just include our css naming conventions with the SCSS style guide?

**Comment by geekman-rohit** [comment]
@kristyjy , yes, and some guidance to use of classes, and other practices we use... no cascading, no specificity, how we name variables, use of jsass etc.
